main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -225,10 +221,24 @@ object CompletionProvider {
 
   /**
     * Characters that constitute a word.
-    * This is more permissive than the parser, but that's OK.
     */
-  private val isWordChar = Letters.LegalLetter ++ Letters.OperatorLetter ++
-    Letters.MathLetter ++ Letters.GreekLetter ++ CharPredicate("@/.")
+  private def isWordChar(c: Char) = isLetter(c) || Lexer.isMathNameChar(c) || Lexer.isGreekNameChar(c) || Lexer.isUserOp(c).isDefined
+
+  /**
+    * Characters that may appear in a word.
+    */
+  private def isLetter(c: Char) = c match {
```

Comment:
Why do we not refer to the Lexer for this part?

Reply:
Ok, but what is the reason we are more permissive?

Reply:
I had to re-implement this function because we are removing `parboilded2` which it depended on. So I just made a function that does the same thing.

I have to admit I don't know why it needs to behave this way. `CompletionProvider` is a but outside my expertise 😅